sms_no_alter_body/models/mail_thread.py

```diff
@@ -0,0 +1,16 @@
+# Copyright (C) 2021 Akretion (<http://www.akretion.com>).
+# @author Kévin Roche <kevin.roche@akretion.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class MailThread(models.AbstractModel):
+    _inherit = "mail.thread"
+
+    @api.returns("mail.message", lambda value: value.id)
+    def _message_sms(self, body, **kwargs):
+        if self._context.get("sms_no_alter_body") is not False:
+            test = self.sudo().with_context(sms_no_alter_body=body)
+            return super(MailThread, test)._message_sms(body, **kwargs)
+        return super()._message_sms(body, **kwargs)
```

Comment:
What we want is to always keep the body when posting a message
The method _message_sms is call in two place in odoo
Here: https://github.com/odoo/odoo/blob/5db838cfc91d8438d6234df00bd8174768445974/addons/sms/models/mail_thread.py#L193
And here : https://github.com/odoo/odoo/blob/5db838cfc91d8438d6234df00bd8174768445974/addons/sms/wizard/sms_composer.py#L243

In both case the body is text and there is no need to alter it, so I see no case where the alteration is needed, so never alter it.

```suggestion
    def _message_sms(self, body, **kwargs):
        self.with_context(force_sms_body=body)
        return super()._message_sms(body, **kwargs)
```

Reply:
You re right